### DISCUSSION TITLE
Will Malt Development continue now that Miguel and Goo Studios are working with Blender for NPR

### DISCUSSION THREAD

**Discussion by yusrig** [bnpr/Malt#596] [General]
Hey so just saw that Goo Studios and even Miguel (who I think is the main dev of Malt, correct me if I am wrong) are going to be working on developing some native NPR tools for blender Foundation. Does this mean Malt will slowly be phased out or continue to run side by side?

Just curious I suppose.

**Comment by pragma37** [comment]
Hi! Sorry for the late reply, I've been away from development stuff for the holidays.

Malt has been in "maintenance mode" for about 2 years now.
It's updated to work on the latest Blender version and for the occasional bug fix, and it will likely stay that way for the foreseeable future.

The original/main goal of Malt was to provide an extensible framework to develop NPR techniques in an "as easy as possible" way, and I consider the project "finished" in that sense.
Any major improvement in that direction would require upgrading to Vulkan to have access to the latest GPU capabilities (like ray tracing).
This is something I toy around with from time to time as I'd personally really like to have it, but Vulkan is so hostile to the "as easy as possible" concept that it's hard to say if that development would lead to a "Malt 2.0" or to something else.

I think if everything goes right with the NPR Blender project, most Malt users using it as an "off the shelf" engine may likely want to switch to it, though.

**Reply by Dereuz24** [reply]
Hey Miguel!

Just wanted to say thank you man. I came here looking to see if anyone mentioned Vulkan because of the imminent release of Blender 4.5 and I found this post

You've completely redefined my workflow and everything I do with your project. With the passing of each day and the need to mimic my 2D style within Blender I ended up understanding your original purpose with Malt.

If it wasn't for you I would've never tried to jump on 3D rendering. The quality of what can be achieved within your project is unbelievable, and the open end of every setting makes it practically limitless in the long run.

**Reply by Dereuz24** [reply]
I also must say that if it wasn't for you I would've been a lot less eager to continue my profession given the whole AI stuff happening out there. It really gives anyone the capability of achieving really high quality top of the line rendering results of something that can look like ANY 2D NPR style. You truly gave me back a lot of control in a situation in which every artist feels it's going to be left out, and a capability of outputting more while still be able to do it all by hand. 

I never managed to see anything close to it with Blender. Maybe the thing that was the closest was the Cinema 4D native because of how versatile it was, but anything else I always found it subpar. Malt is truly magnificent, it's the first time I'm able to see on 3D something that could've been done with traditional paper and pencil 2D animators.

Just wanted to say these things after reading your comment. Thank you and also thanks a lot for considering keeping Malt in maitenance/compatibility until NPR becomes a suitable replacement.

Even if NPR doesn't become that, the fact that every couple of years we have better hardware a lot of the things I  thought were impossible are available right now even with how heavy is to run sometimes!

I wish Malt could run on GPU with Vulkan, but even in this state is everything I need!

PS: At least the performance improvements will allow us to use eevee next more effectively in combination with Malt for the backgrounds and scenes that require it. 4.1 has been a must for a lot of the requirements these projects have because of the jump to Eevee next but if things get more performant, who is it to say that we aren't allowed to keep using Malt in combination with the NPR project in compositing?

If you are still maintaining Malt in the future, even if we don't get new features, it is so powerful and customizable that I think it would be amazing to combine with all the other engines for whenever it is needed and for anything that can't be achieved with Eevee, it would be the best of both worlds!

Thanks a lot Miguel!

**Reply by eballeste** [reply]
Im curious, what are your thoughts on the new direction that Blender is taking with NPR? How does Malt fit in with the upcoming Blender 5.0?
